gym_minigrid/envs/dynamic_minigrid.py

```diff
@@ -90,22 +132,43 @@ def alter_goal_pos():
                 self.goal_pos = new_goal_pos  # change the attribute
                 self.grid.set(*new_goal_pos, Goal())  # change the actual element in the grid
                 self.grid.set(*goal_pos, None)  # remove the previous goal
+                if not is_solvable():
```

Comment:
you're checking `is_solvable()` in the condition for the while loop and here, which means twice per iteration, right? maybe you can reduce this to one? also i think the structure of the loop between the new start/goal/object should be the same, no?